File: reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int check =1;
        if(x>0)
            check=1;
        else
        {
            check=-1;
            x = x*-1;
        }
        long res=0;
        
        while(x>0)
        {
            int temp = x%10;
            res = res*10+temp;
            x=x/10;
        }
        if((res*check)<-2147483648 | (res*check)>2147483647)
            return 0;
        else
            return (int)res*check;
    }
}

File: reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        temp = str(x)
        ck =False
        if temp[0]=='-':
            temp = temp[1:]
            ck=True
        temp = temp[::-1]
        if ck:
            temp='-'+temp
        if int(temp) > math.pow(2,31)-1 or int(temp) < math.pow(2,31)*-1:
            return 0;
        return int(temp)
        